src/ng/directive/input.js

```diff
@@ -1290,6 +1290,9 @@ function createDateParser(regexp, mapping) {
           }
         });
         return new Date(map.yyyy, map.MM - 1, map.dd, map.HH, map.mm, map.ss || 0, map.sss * 1000 || 0);
+      } else {
+        // If we fail to parse the string with the ISO regex, it might be because it's the result of Date.toString()
+        return Date.parse(iso);
```

Comment:
Yes, this is only the make it work with Date.toString(). Maybe we can only use Date.parse() when the string has a minimum length of x?